exonum/src/runtime/error.rs

```diff
@@ -522,16 +1159,127 @@ mod tests {
         ];
 
         for value in values {
-            let res = ExecutionStatus(value.map_err(|(kind, description)| ExecutionError {
-                kind,
-                description: description.to_owned(),
-            }));
-            let body = serde_json::to_string_pretty(&res).unwrap();
-            let res2 = serde_json::from_str(&body).unwrap();
+            let mut res =
+                ExecutionStatus(value.map_err(|(kind, description)| {
+                    ExecutionError::new(kind, description.to_owned())
+                }));
+            let json = serde_json::to_string_pretty(&res).unwrap();
+            let res2 = serde_json::from_str(&json).unwrap();
             assert_eq!(res, res2);
+
+            if let Err(err) = res.0.as_mut() {
+                err.runtime_id = Some(1);
+                let json = serde_json::to_string_pretty(&res).unwrap();
+                let res2 = serde_json::from_str(&json).unwrap();
+                assert_eq!(res, res2);
+            }
+
+            if let Err(err) = res.0.as_mut() {
+                err.call_site = Some(CallSite {
+                    instance_id: 1_000,
+                    call_type: CallType::BeforeCommit,
+                });
+                let json = serde_json::to_string_pretty(&res).unwrap();
+                let res2 = serde_json::from_str(&json).unwrap();
+                assert_eq!(res, res2);
+            }
+
+            if let Err(err) = res.0.as_mut() {
+                err.call_site = Some(CallSite {
+                    instance_id: 1_000,
+                    call_type: CallType::Method {
+                        interface: "exonum.Configure".to_owned(),
+                        id: 1,
+                    },
+                });
+                let json = serde_json::to_string_pretty(&res).unwrap();
+                let res2 = serde_json::from_str(&json).unwrap();
+                assert_eq!(res, res2);
+            }
         }
     }
 
+    #[test]
+    #[allow(clippy::cognitive_complexity)] // More test code is fine
+    fn execution_error_matching() {
+        let mut error = ExecutionError {
+            kind: ErrorKind::Unexpected,
+            description: "Panic!".to_string(),
+            runtime_id: None,
+            call_site: None,
+        };
+        let mut matcher = ExecutionErrorMatch {
+            kind: ErrorKind::Unexpected,
+            description: StringMatch::Any,
+            runtime_id: None,
+            instance_id: None,
+            call_type: None,
+        };
+        assert_eq!(error, matcher);
+
+        // Check various description types.
+        matcher.description = StringMatch::Exact("Panic!".to_owned());
```

Comment:
You can simplify code like this by the `vec![(StringMatch::Exact("Panic!".to_owned()), true)]`

Reply:
I think this improvement would increase the amount of code 🙃